lib/Mojo/Message.pm

```diff
@@ -99,10 +99,20 @@ sub fix_headers {
   my $headers = $content->headers;
   if ($content->is_multipart) { $headers->remove('Content-Length') }
   elsif ($content->is_chunked || $headers->content_length) { return $self }
-  if   ($content->is_dynamic) { $headers->connection('close') }
-  else                        { $headers->content_length($self->body_size) }
-
-  return $self;
+  if ($content->is_dynamic) { 
+    $headers->connection('close') 
+  }
+  else {
+    my $size = $self->body_size;
+    my $is_websocket = defined $headers->sec_websocket_version || defined $headers->sec_websocket_accept; 
+    if (!$is_websocket || $size > 0) {
+      $headers->content_length($size) 
+    }
+    else { 
+      $headers->remove('Content-Length')
```

Comment:
Do you need to remove the Content-Length header?

Reply:
that's the whole point here. to get rid of it

Reply:
But is the header there at this point? I thought `content_length()` was not called in this case...